server/util_test.go

```diff
@@ -253,183 +253,203 @@ func mustDecodeStr(c *C, b []byte) string {
 
 func (s *testUtilSuite) TestAppendFormatFloat(c *C) {
 	infVal, _ := strconv.ParseFloat("+Inf", 64)
+
+	float64FromTableDefaultWidth := ColumnInfo{
+		Table:        "fromTable",
+		ColumnLength: 22,
+		Decimal:      31,
+	}
+
+	float32FromTableDefaultWidth := ColumnInfo{
+		Table:        "fromTable",
+		ColumnLength: 12,
+		Decimal:      31,
+	}
+
 	tests := []struct {
-		fVal    float64
-		out     string
-		prec    int
-		bitSize int
+		fVal       float64
+		out        string
+		columnInfo ColumnInfo
+		bitSize    int
 	}{
 		{
 			99999999999999999999,
 			"1e20",
-			-1,
+			float64FromTableDefaultWidth,
 			64,
 		},
 		{
 			1e15,
 			"1e15",
-			-1,
+			float64FromTableDefaultWidth,
+			64,
+		},
+		{
+			1e15,
+			"1e15",
+			ColumnInfo{
+				Table:        "",
+				ColumnLength: 22,
+				Decimal:      31,
+			},
 			64,
 		},
 		{
 			9e14,
 			"900000000000000",
-			-1,
+			float64FromTableDefaultWidth,
 			64,
 		},
 		{
 			-9999999999999999,
 			"-1e16",
-			-1,
+			float64FromTableDefaultWidth,
 			64,
 		},
 		{
 			999999999999999,
 			"999999999999999",
-			-1,
+			float64FromTableDefaultWidth,
 			64,
 		},
 		{
 			0.000000000000001,
 			"0.000000000000001",
-			-1,
+			float64FromTableDefaultWidth,
 			64,
 		},
 		{
 			0.0000000000000009,
 			"9e-16",
-			-1,
+			float64FromTableDefaultWidth,
 			64,
 		},
 		{
 			-0.0000000000000009,
 			"-9e-16",
-			-1,
-			64,
-		},
-		{
-			0.11111,
-			"0.111",
-			3,
-			64,
-		},
```

Comment:
I deleted this test because for a column of 3 decimals, the input is already rounded when we store it. So the result is correct in an integration test.
However the result from the function test will not be correct. Let me know if you want me to implement this in the output formatter too, even though its truncated on input.